src/main/java/fi/oph/akt/repository/LanguagePairRepository.java

```diff
@@ -14,8 +14,17 @@
 			+ " FROM LanguagePair lp")
 	List<AuthorisationLanguagePairProjection> listAuthorisationLanguagePairProjections();
 
-	@Query("SELECT new fi.oph.akt.repository.TranslatorLanguagePairProjection(t.id, lp.fromLang, lp.toLang) FROM LanguagePair lp"
-			+ " JOIN lp.authorisation a JOIN a.translator t WHERE lp.permissionToPublish=true AND t.id IN ?1")
+	// @formatter:off
+	@Query("SELECT new fi.oph.akt.repository.TranslatorLanguagePairProjection(t.id, lp.fromLang, lp.toLang)" +
```

Comment:
Nyt kun tämä kysely kutakuinkin vastaan TranslatorRepositoryn vastaavaa kyselyä, niin olisiko sama asia hakea ensin kaikkien kääntäjien kaikki kieliparit tällä kyselyllä (jättäen viimeisen rivin kyselystä pois) ja sen jälkeen listata kääntäjät metodilla `translatorRepository.findAllById` (joka ottaisi vastaan listan kääntäjien id:eitä jotka saadaan täältä projektiosta) ?

Reply:
Siitähän tuli näin vähän nopeampikin.